File: src/main/java/nl/crosscode/x509certbuddy/utils/X509Utils.java
package nl.crosscode.x509certbuddy.utils;

import java.security.cert.CertificateEncodingException;
import java.security.cert.X509Certificate;
import java.util.Base64;

public class X509Utils {

    public static String getBase64(X509Certificate cert) {
        try {
            return Base64.getEncoder().encodeToString(cert.getEncoded());
        } catch (CertificateEncodingException e) {
            return null;
        }
    }
}

File: src/main/kotlin/nl/crosscode/x509certbuddy/utils/X509Utils.kt
package nl.crosscode.x509certbuddy.utils

import java.security.cert.X509Certificate
import kotlin.io.encoding.Base64
import kotlin.io.encoding.ExperimentalEncodingApi

@OptIn(ExperimentalEncodingApi::class)
fun getBase64(cert: X509Certificate): String {
    return Base64.encode(cert.encoded)
}
